### PR TITLE
[BREAKING] Redo path specifications

### LINKED ISSUE #93: allow for "releases only" mode and may be RF specification for the paths?
I wanted to test my ability to RTFM into its details on either I need to specify `type_id` to place releases into subfolders so came up with this
```
repo: datalad/git-annex
vars:
  common_path: 'builds/{year}/{month}//{type}-{type_id[:8]}/{wf_file}-{number}-{build_commit[:8]}-{common_status}/'
ci:
  github:
    # only releases
    releases_path: 'releases/{type_id}'
    workflows:
    - build-macos.yaml
    - build-ubuntu.yaml
    - build-windows.yaml
since: 2021-03-01T00:00:00Z
types: []
```
config but running it crashed with
```
Traceback (most recent call last):
  File "/home/datalad/miniconda3/envs/tinuous-dev/bin/tinuous", line 33, in <module>
    sys.exit(load_entry_point('tinuous', 'console_scripts', 'tinuous')())
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 1256, in invoke
    Command.invoke(self, ctx)
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/mnt/datasets/datalad/ci/tinuous/src/tinuous/__main__.py", line 60, in main
    ctx.obj = Config.parse_obj(safe_load(fp))
  File "pydantic/main.py", line 578, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 406, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Config
ci -> github -> path
  field required (type=value_error.missing)
```
suggesting that `path` (for logs) is mandatory.  Now that we have different `path`s and ppl might want or not download just releases or just artifacts, I think we should allow for path to be optional.

I even wonder if it would be worthwhile to RF into
```
paths:
  logs:
  artifacts:
  releases:
```
to make specification cleaner (good that we don't have yet too many deployments to adjust config for ;))

### PR DESCRIPTION
**Pull request by jwodder**
Closes #93.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -184,20 +190,28 @@ keys:
...
     ``appveyor``
         Configuration for retrieving logs from Appveyor.  Subfields:
 
-        ``path``
-            *(required)* A template string that will be instantiated for each
-            job of each build to produce the path for the file (relative to the
-            current working directory) in which the job's logs will be saved.
-            See "`Path Templates`_" for more information.
+        ``paths``
+            *(required)* A mapping giving `template strings <Path Templates_>`_
```
Comment (yarikoptic) [inline README.rst:207]: It isn't required any longer or is it?

Comment (jwodder) [reply README.rst:207]: It is required, but it can be empty.

Comment (yarikoptic) [reply README.rst:207]: would making it truly optional be doable?  it is kinda odd to allow for it to be empty as indicator of "optional"

Comment (jwodder) [reply README.rst]: It's now optional.

**Review by yarikoptic — APPROVED** [review]

**Pull request merged**